### PR TITLE
match_matches_window: Check if focused->window exists

### LINKED ISSUE #3406: Crash on parsing for_window config
## I'm submitting a…

<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior

Adding the line

    for_window [class="__focused__" instance="virt-manager"] exec xdotool mousemove 0 0

will crash i3 upon loading the config.

## Expected Behavior

Startup with nagbar notifying me about an error in my config.

## Reproduction Instructions

Add `for_window [class="__focused__" instance="virt-manager"] exec xdotool mousemove 0 0` to the config and restart in-place.

EDIT:
Adding `for_window [class="__focused__"] floating enable` (or any other command) will also result in a crash.

## Environment

Output of `i3 --moreversion 2>&-`:
<pre>
i3 version: 
Binary i3 version:  4.15-258-g18f705a9 (2018-09-14, branch "next") © 2009 Michael Stapelberg and contributors
Running i3 version: 4.15-258-g18f705a9 (2018-09-14, branch "next") (pid 4526)
Loaded i3 config: /home/martin/.config/i3/config (Last modified: Sat 15 Sep 2018 01:09:12 AM CEST, 275 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
</pre>

Default config with Alt as $mod and the abovementioned line added near the end:
https://pastebin.com/bbySQTJu

<pre>
Logfile: No logfile since it crashes on start
Backtrace: `No stack.`
</pre>

<pre>
- Linux Distribution & Version: Arch Linux, 4.18.7-arch1-1-ARCH
- Are you using a compositor (e.g., xcompmgr or compton): None
</pre>

### PR DESCRIPTION
**Pull request by orestisf1993**
Fixes #3406.

### CODE REVIEW

File: src/match.c [hunk 1]
```diff
@@ -87,31 +87,30 @@ void match_copy(Match *dest, Match *src) {
...
-        if (strcmp(match->class->pattern, "__focused__") == 0 &&
-            strcmp(window->class_class, focused->window->class_class) == 0) {
-            LOG("window class matches focused window\n");
-        } else if (regex_matches(match->class, window->class_class)) {
-            LOG("window class matches (%s)\n", window->class_class);
-        } else {
-            return false;
-        }
-    }
+#define GET_FIELD_0(field) (field)
```
Comment (Airblader) [inline src/match.c]: I don't like the numeric suffix, it doesn't really convey meaning at all. I'd rather add a suffix that better conveys meaning.

File: src/match.c [hunk 2]
```diff
@@ -87,31 +87,30 @@ void match_copy(Match *dest, Match *src) {
...
+#define GET_FIELD_1(field) (i3string_as_utf8(field))
+#define CHECK_WINDOW_FIELD(match_field, window_field, is_i3_string)                                       \
```
Comment (Airblader) [inline src/match.c]: These macros should also be undefined again in the end

File: src/match.c (hunk 1 above)
Comment (orestisf1993) [reply src/match.c]: Better now?

File: src/match.c (hunk 2 above)
Comment (orestisf1993) [reply src/match.c]: Not really necessary though, not going to affect anything else.

**Pull request merged**